src/broker/redis.rs

```diff
@@ -41,95 +46,187 @@ impl BrokerBuilder for RedisBrokerBuilder {
+    fn new(connection: ConnectionManager, queue_name: String) -> Self {
+        Self {
+            connection,
+            queue_name,
+        }
+    }
+
+    fn pending_queue(&self) -> String {
+        format!("_celery.{}_pending", self.queue_name)
+    }
+
+    fn process_map_name(&self) -> String {
+        format!("_celery.{}_process_map", self.queue_name)
+    }
+
+    async fn fetch_task(
+        mut self,
+        send_waker: Option<(Sender<Waker>, Waker)>,
+    ) -> Result<Delivery, BrokerError> {
+        let pending_queue = self.pending_queue();
+        if let Some((mut sender, waker)) = send_waker {
+            sender.send(waker).await.unwrap();
+            futures::pending!();
+        }
+        loop {
+            let rez: Result<Option<String>, RedisError> = redis::cmd("RPOPLPUSH")
+                .arg(&self.queue_name)
+                .arg(&pending_queue)
+                .query_async(&mut self.connection)
+                .await;
+            match rez {
+                Ok(None) => tokio::time::delay_for(tokio::time::Duration::from_millis(1000)).await,
+                Ok(Some(rez)) => {
+                    let delivery: Delivery = serde_json::from_str(&rez[..])?;
+                    debug!(
+                        "Received msg: {} / {}",
+                        delivery.delivery_tag, delivery.headers.task
+                    );
+                    let _set_rez: u32 = redis::cmd("HSET")
+                        .arg(&self.process_map_name())
+                        .arg(&delivery.correlation_id)
+                        .arg(&rez)
+                        .query_async(&mut self.connection)
+                        .await?;
+                    let _lrem_rez: u32 = redis::cmd("LREM")
+                        .arg(&pending_queue)
+                        .arg(-1)
+                        .arg(&rez)
+                        .query_async(&mut self.connection)
+                        .await?;
+                    break Ok(delivery);
+                }
+                Err(err) => break Err(err.into()),
+            }
+        }
+    }
+
+    async fn send_task(mut self, message: &Message) -> Result<(), BrokerError> {
+        Ok(redis::cmd("LPUSH")
+            .arg(&self.queue_name)
+            .arg(message.json_serialized()?)
+            .query_async(&mut self.connection)
+            .await?)
+    }
+
+    async fn send_task_front(mut self, message: &Message) -> Result<(), BrokerError> {
+        Ok(redis::cmd("RPUSH")
+            .arg(&self.queue_name)
+            .arg(message.json_serialized()?)
+            .query_async(&mut self.connection)
+            .await?)
+    }
+
+    async fn retry_task(&self, delivery: &Delivery) -> Result<(), BrokerError> {
+        let mut message = delivery.clone().try_deserialize_message()?;
+        let retries = message.headers.retries.unwrap_or_default();
+        message.headers.retries = Some(retries + 1);
+        self.clone().send_task_front(&message).await?;
```

Comment:
Following kombu, the task goes back to end of queue I think. 
```suggestion
        self.clone().send_task(&message).await?;
```